### PR TITLE
tests: call file directly for the HasArchitecture checker

### PR DESCRIPTION
**Pull request by sergiusens**
Signed-off-by: Sergio Schvezov <sergio.schvezov@canonical.com>

- [x] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [x] If this is a bugfix. Have you checked that there is a bug report open for the issue you are trying to fix on [bug reports](https://bugs.launchpad.net/snapcraft)?
- [x] If this is a new feature. Have you discussed the design on the [forum](https://forum.snapcraft.io)?
- [x] Have you successfully run `./runtests.sh static`?
- [x] Have you successfully run `./runtests.sh unit`?

-----

### CODE REVIEW

File: snapcraft/tests/matchers.py [hunk 1]
```diff
@@ -23,20 +23,29 @@ class HasArchitecture:
...
+                'Failed to get architecture info: {}', cmd_err) from cmd_err
+
+        try:
+            # output is of the form
+            # ELF 64-bit LSB executable, ARM aarch64, version 1 (SYSV), \
+            # dynamically linked, interpreter /lib/ld-linux-aarch64.so.1, \
+            # for GNU/Linux 3.7.0, \
+            # BuildID[sha1]=58b8495bef7a6c9e5f403879e29dc63ad4ed2d11, \
+            # not stripped
+            arch = output.split(',')[1]
```
Comment (kyrofa) [inline snapcraft/tests/matchers.py]: On amd64 I get:

`<file name>: ELF 64-bit LSB executable, x86-64, version 1 (GNU/Linux), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, for GNU/Linux 2.6.32, BuildID[sha1]=a4b02c379f4482e7cb3148e0d3fc9459a9126cfd, not stripped`

`output.split(',')[1]` is ` x86-64` (notice the space). Might want to strip this.

File: snapcraft/tests/matchers.py [hunk 2]
```diff
@@ -23,20 +23,29 @@ class HasArchitecture:
...
         return 'HasArchitecture()'
 
     def match(self, file_path):
-        magic = self._ms.file(file_path)
-        # Catch exceptions on splitting the string to provide context
-        # This includes "cannot open `...' (No such file or directory)"
+        cmd = ['file', file_path]
         try:
-            arch = magic.split(',')[1]
+            output = subprocess.check_output(cmd).decode()
```
Comment (kyrofa) [inline snapcraft/tests/matchers.py]: Should you be specifying an encoding, here?

Comment (sergiusens) [reply snapcraft/tests/matchers.py]: The default is utf-8 and from what I've read is sane enough (and common practice)

Comment (sergiusens) [reply snapcraft/tests/matchers.py]: which we required btw due to use of click

File: snapcraft/tests/matchers.py (hunk 1 above)
Comment (sergiusens) [reply snapcraft/tests/matchers.py]: the original implementation is exactly the same, except that we call `file` directly.

This is why we do `expected_arch in arch`. Notice that on arm64 `output.split(',')[1]` is `ARM aarch64`

Comment (kyrofa) [reply snapcraft/tests/matchers.py]: Ah yes!

**Pull request merged**